In [1]:
 import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

# Не показывать FutureWarnings
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
 # импорт данных
df = pd.read_csv('sleep75.csv')

In [5]:
 mod = smf.ols(formula = 'sleep ~ 1+totwrk+male+marr+south' , data = df)
res = mod.fit()

In [9]:
# R^2
res.rsquared.round(3)

0.123

In [8]:
 # adj R^2
res.rsquared_adj.round(3)

0.118

$$ sleep = \beta_0 + \beta_1 totwork + \beta_2 male + \beta_3 marr + \beta_4 south + u$$

Коэффициент детерминации равен 0.123. Такая доля показателя длительности сна (в минутах) зависит от общей занятости в неделю (в минутах), от бинарного показателя пола (1 - мужчина), от бинарного показателя состояния в браке (1 - в браке) и от бинарного показателя географического положения (1 - на юге)